napi.h

```diff
@@ -72,51 +80,63 @@ namespace Napi {
-   */
+  /// A JavaScript value of unknown type.
+  ///
+  /// For type-specific operations, convert to one of the Value subclasses using a `To*` or `As*`
+  /// method. The `To*` methods do type coercion; the `As*` methods do not.
   class Value {
   public:
-    Value();
-    Value(napi_env env, napi_value value);
+    Value(); ///< Creates a new _empty_ Value instance.
+    Value(napi_env env, napi_value value); ///< Wraps a N-API value primitive.
 
+    /// Converts to a N-API value primitive.
+    ///
+    /// If the instance is _empty_, this returns `nullptr`.
     operator napi_value() const;
+
+    /// Tests if this value strictly equals another value.
     bool operator ==(const Value& other) const;
+
+    /// Tests if this value does not strictly equal another value.
     bool operator !=(const Value& other) const;
+
+    /// Tests if this value strictly equals another value.
     bool StrictEquals(const Value& other) const;
 
+    /// Gets the environment the value is associated with.
     Napi::Env Env() const;
 
-    napi_valuetype Type() const;
-    bool IsUndefined() const;
-    bool IsNull() const;
-    bool IsBoolean() const;
-    bool IsNumber() const;
-    bool IsString() const;
-    bool IsSymbol() const;
-    bool IsArray() const;
-    bool IsArrayBuffer() const;
-    bool IsTypedArray() const;
-    bool IsObject() const;
-    bool IsFunction() const;
-    bool IsBuffer() const;
-
-    // Use As<T> to convert to another type of Napi::Value, when the actual type is
-    // known or assumed. This conversion does NOT very the type. But calling any methods
-    // inappropriate for the actual value type will cause Napi::Errors to be thrown.
+    napi_valuetype Type() const; ///< Gets the type of the value.
+
+    bool IsUndefined() const;   ///< Tests if a value is an undefined JavaScript value.
+    bool IsNull() const;        ///< Tests if a value is a null JavaScript value.
+    bool IsBoolean() const;     ///< Tests if a value is a JavaScript boolean.
+    bool IsNumber() const;      ///< Tests if a value is a JavaScript number.
+    bool IsString() const;      ///< Tests if a value is a JavaScript string.
+    bool IsSymbol() const;      ///< Tests if a value is a JavaScript symbol.
+    bool IsArray() const;       ///< Tests if a value is a JavaScript array.
+    bool IsArrayBuffer() const; ///< Tests if a value is a JavaScript array buffer.
+    bool IsTypedArray() const;  ///< Tests if a value is a JavaScript typed array.
+    bool IsObject() const;      ///< Tests if a value is a JavaScript object.
+    bool IsFunction() const;    ///< Tests if a value is a JavaScript function.
+    bool IsBuffer() const;      ///< Tests if a value is a Node buffer.
+
+    /// Converts to another type of `Napi::Value`, when the actual type is known or assumed.
+    ///
+    /// This conversion does NOT coerce the type. Calling any methods inappropriate for the actual
+    /// value type will cause `Napi::Error`s to be thrown.
```

Comment:
Just nitpicky, should we write this as such?

Calling any methods inappropriate for the actual value type will throw `Napi::Error`.